src/components/NewTodoField/NewTodoField.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from 'react';
+
+interface Props {
+  newTodoField: React.RefObject<HTMLInputElement>;
+  isAdding: boolean;
+  addNewTodo: (title: string) => void;
+  setErrorMessage: (message: string) => void;
+  setHasError: (val: boolean) => void;
+  changeAllToCompleted: () => void;
+}
+
+export const NewTodoField: React.FC<Props> = ({
+  newTodoField,
+  isAdding,
+  addNewTodo,
+  setErrorMessage,
+  setHasError,
+  changeAllToCompleted,
+}) => {
+  const [title, setTitle] = useState('');
+
+  const handleSubmit = (event : React.FormEvent<HTMLElement>) => {
+    event.preventDefault();
+
+    if (title.trim() === '') {
+      setHasError(true);
+      setErrorMessage('Title can\'t be empty');
+
+      return;
+    }
+
+    addNewTodo(title);
+    setTitle('');
+  };
+
+  return (
+    <header className="todoapp__header">
+      {/* eslint-disable jsx-a11y/control-has-associated-label */}
+      <button
+        data-cy="ToggleAllButton"
```

Comment:
https://user-images.githubusercontent.com/110383794/201534211-9f95ef6e-ef68-48fa-8568-c00f7e76f331.mp4

ToggleAll button should change every todo status to not completed after clicking, and the loader should occur on every todo while status changes.